flink-ml-lib/src/main/java/org/apache/flink/ml/feature/tokenizer/Tokenizer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.tokenizer;
+
+import org.apache.flink.ml.api.Transformer;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.util.ParamUtils;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.table.api.Expressions;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.functions.ScalarFunction;
+import org.apache.flink.util.Preconditions;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.table.api.Expressions.$;
+
+/**
+ * A tokenizer is a transformer that converts the input string to lowercase and then splits it by
+ * white spaces.
+ */
+public class Tokenizer implements Transformer<Tokenizer>, TokenizerParams<Tokenizer> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public Tokenizer() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+
+        Expression tokenizerUdf = Expressions.call(TokenizerUdf.class, $(getInputCol()));
+        Table output = inputs[0].addColumns(tokenizerUdf).as(getInputCol(), getOutputCol());
```

Comment:
Could you please verify that the current implementation preserves columns that are not `getInputCol()` after this transformation? I'm a little concerned that if there are more than two input columns for this transformer, exceptions would be thrown.

Reply:
Good catch. I will update the code and the corresponding test here.